Sources/BraintreeCore/BTAnalyticsService.swift

```diff
@@ -1,5 +1,11 @@
 import Foundation
 
+/// :nodoc: Enum used for status to send to analytics event
+@objc public enum BTAnalyticsResult: Int {
```

Comment:
We don't know if this will be applicable to all payment methods. For example, some have cancel cases, some don't. Also there might be one off completion types like `paypalShippingCallback` called or something.

So maybe a global enum for all payment method modules won't fit our needs?

Reply:
I think it could make sense for the tense of the common cases (started/network connection lost/failed/succeeded) but then we could still handle edge cases on a case by case basis. Again, if we feel strongly against this, we can remove this but thought it worth discussing.

Reply:
whatever we do, I'm not a fan of associated values

Reply:
Is there a specific reason you are not a fan of associated values, @KunJeongPark?

Reply:
+1 - associated values are really powerful, curious too why you say that?

Reply:
It depends for me but there are definitely some cases where associated values get [weird](https://stackoverflow.com/a/32154586). A lot of gymnastics is required to work with them when outside of a `switch` statement.

Reply:
Yes, for someone seeing it for first time, it can be a bit confusing. You can have different data types as associated values, having to click in to see what type it is. You can't iterate over all cases (which we don't need in our case)  and I think changes are sometimes harder to reason about.

Reply:
> It depends for me but there are definitely some cases where associated values get [weird](https://stackoverflow.com/a/32154586). A lot of gymnastics is required to work with them when outside of a switch statement.

This is an odd example for sure - typically you'd do something like the function `doToLine` in an extension on the enum (in later versions of Swift).

> You can't iterate over all cases (which we don't need in our case)

🤔  I am not sure this is true. This works as expected in a playground - it just cannot automatically synthesize `allCases` so you need to provide it:
```
enum Cases: CaseIterable {
    static var allCases: [Cases] {
        [.one(""), .two(""), .three("")]
    }

    case one(String)
    case two(String)
    case three(String)
}

Cases.allCases.forEach { print($0) }
```

Agree they can get complicated once you start passing in tuples, a ton of types, etc that aren't easily able to be reasoned about when looking through the callsite code (vs simple types - String, Error, Int, etc). Also agree they can get weird, but I do think that they are a powerful language feature when used correctly and don't want to discount them entirely in our codebases when they are the right language tool to use for a given case.